x/nexus/keeper/chain.go

```diff
@@ -32,7 +37,7 @@ func (k Keeper) getChainState(ctx sdk.Context, chain exported.Chain) (chainState
 }
 
 // RegisterAsset indicates that the specified asset is supported by the given chain
-func (k Keeper) RegisterAsset(ctx sdk.Context, chain exported.Chain, asset exported.Asset) error {
+func (k Keeper) RegisterAsset(ctx sdk.Context, chain exported.Chain, asset exported.Asset, limit sdk.Int) error {
```

Comment:
```suggestion
func (k Keeper) RegisterAsset(ctx sdk.Context, chain exported.Chain, asset exported.Asset, limit sdk.Uint) error {
```
so that you don't have to check `IsNegative`

Reply:
Yes, the interface should pick the type that makes the most sense and then the conversion can happen internally in a safer way.